src/tv2-common/parts/server.ts

```diff
@@ -37,9 +37,18 @@ export interface ServerPartProps {
 	tapeTime: number
 	adLibPix: boolean
 	session?: string
-	lastServerPosition?: ServerPosition
 	actionTriggerMode?: ServerSelectMode
+	lastServerPosition?: ServerPosition
+}
+
+export interface ServerContentProps {
 	seek?: number
+	/** Clip duration from mediaObject or tapeTime */
+	clipDuration?: number
+	/** Clip duration as in `clipDuration` but with postroll subtracted */
+	sourceDuration?: number
```

Comment:
Is it necessary that `clipDuration` and `sourceDuration` are optional?

I would like us to minimize our use of optional types and ` | undefined`. The reason for this is that everytime we declared a variable to be optional or that it can be undefined, we enforce that we write (unecessary) null- /undefined-checks throughout our codebase where these variables are used.

Offcourse, sometimes it's needed and if this is one of those cases it's fine. I just want us to think more about our use of them and try to design our code in such a way, that it's not needed.

Reply:
At the very end these are passed to objects that allow `undefined`. But I changed them to be  `| undefined` to enforce them being explicitly set.

Reply:
I know that we have a lot of places where we allow stuff to be undefined. That is why I would like us to start thinking of our use of it, so we slowly can get rid of it. The more we use it, the more places we are going to add it since the varaibles we are dealing with all can be undefined.